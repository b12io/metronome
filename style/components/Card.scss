.ds-card {
  border: $ds-input-border;
  border-radius: $ds-border-radius-base;
  box-shadow: 0 1px 2px 0 rgba(17, 15, 51, 0.1);
  padding: 8px 36px 8px 10px;
  position: relative;
  cursor: pointer;
  display: block;
  width: 100%;
  background: #fff;
  text-align: left;
  transition: $ds-transition;

  + .ds-card {
    margin-top: 10px;
  }

  &__icon {
    position: absolute;
    top: 10px;
    right: 10px;
    width: 16px;
    height: 16px;
    text-align: center;

    svg,
    path {
      transition: $ds-transition;
    }
  }

  &__label {
    font-family: $ds-font-family-base;
    font-size: 12px;
    line-height: 18px;
    color: $ds-body-light-color;
  }

  &__value {
    font-size: 14px;
    color: $ds-display-regular-color;
  }

  &--add {
    .ds-card__icon {
      top: 50%;
      transform: translate(0,-50%);
      height: unset;
    }

    .ds-card__value {
      opacity: .2
    }
  }

  &:hover {
    border-color: $ds-ui-darkest-color;

    .ds-card__icon {
      path {
        fill: $ds-brand-light-color;
      }
    }
  }

  &--selected,
  &--selected:hover {
    border-color: $ds-input-border-focus-color;
    box-shadow: inset 0 1px 3px 0 rgba(0, 0, 0, 0.1), inset 0px 0px 0px 1px $ds-input-border-focus-color;
    outline: none;
  }

  &:focus,
  &:active {
    border-color: $ds-input-border-focus-color;
    box-shadow: inset 0 1px 3px 0 rgba(0, 0, 0, 0.1), inset 0px 0px 0px 1px $ds-input-border-focus-color;
    outline: none;

    .ds-card__label {
      color: $ds-brand-light-color;
    }
  }

  &[disabled] {
    &,
    &:hover,
    &:focus,
    &:active {
      border-color: $ds-ui-darker-color;
      box-shadow: none;
      background: $ds-ui-regular-color;

      .ds-card__label,
      .ds-card__value {
        color: $ds-body-light-color;
      }
    }
  }

  &--has-image {
    padding: spacing('xxsm') spacing('lg') spacing('xxsm') spacing('xxsm');

    > img {
      width: 100%;
      border-radius: $ds-border-radius-base;
    }
  }

  &--has-colors {
    padding: spacing('xxsm') spacing('lg') spacing('xxsm') spacing('xxsm');

    .ds-palette-option {
      width: 100%;
      flex-direction: row;
      flex-wrap: nowrap;
      padding: 0;
      margin: 0;
      height: 50px;

      &:hover {
        transform: none !important;
      }
    }
  }
}
