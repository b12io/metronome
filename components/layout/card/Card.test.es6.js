/* global expect */
import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Card from './Card.es6.js'
import { ChevronRight, PlusCircle } from '../../Icons.es6.js'

it('renders the card component', () => {
  const component = shallow(<Card />)
  expect(component.prop('className')).toEqual('ds-card')
})

it('adds the proper class', () => {
  const component = shallow(<Card add />)
  expect(component.prop('className')).toEqual(
    expect.stringContaining('ds-card--add')
  )
})

it('renders correctly', () => {
  const tree = renderer
    .create(<Card icon={<ChevronRight />} label="Link" value="Test Link" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with add prop', () => {
  const tree = renderer
    .create(<Card add icon={<PlusCircle />} label="Link" value="None" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('calls onMouseEnter', () => {
  const onMouseEnter = jest.fn()
  const component = shallow(<Card onMouseEnter={onMouseEnter} />)
  component.find('.ds-card').at(0).simulate('mouseenter')
  expect(onMouseEnter.mock.calls.length).toBe(1)
})

it('calls onMouseEnter', () => {
  const onMouseLeave = jest.fn()
  const component = shallow(<Card onMouseLeave={onMouseLeave} />)
  component.find('.ds-card').at(0).simulate('mouseleave')
  expect(onMouseLeave.mock.calls.length).toBe(1)
})
