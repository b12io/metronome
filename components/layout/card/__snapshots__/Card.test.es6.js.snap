// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="ds-card"
  disabled={false}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="ds-card__icon"
  >
    <svg
      className=""
      height={16}
      viewBox="0 0 7 11"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.8 5.4l-5.4 5.4L0 9.4l4-4-4-4L1.4 0"
        fill="#d8d8d8"
        fillRule="nonzero"
      />
    </svg>
  </div>
  <div
    className="ds-card__label"
  >
    Link
  </div>
  <div
    className="ds-card__value"
  >
    Test Link
  </div>
</button>
`;

exports[`renders correctly with add prop 1`] = `
<button
  className="ds-card ds-card--add"
  disabled={false}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="ds-card__icon"
  >
    <svg
      className=""
      height={16}
      viewBox="0 0 16 16"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm4 9H9v3H7V9H4V7h3V4h2v3h3v2z"
        fill="#d8d8d8"
        fillRule="nonzero"
      />
    </svg>
  </div>
  <div
    className="ds-card__label"
  >
    Link
  </div>
  <div
    className="ds-card__value"
  >
    None
  </div>
</button>
`;
